gcloud-java-contrib/gcloud-java-nio/src/test/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProviderTest.java

```diff
@@ -346,85 +388,101 @@ public void testWrite_absoluteObjectName_disableStrip_slashGetsPreserved() throw
+          }
+        }
+        try {
+          Files.size(path);
+          // we shouldn't make it to this line. Not using thrown.expect because
+          // I still want to run a few lines after the exception.
+          assertThat(false).isTrue();
+        } catch (NoSuchFileException nsf) {
+          // that's what we wanted, we're good.
         }
       }
-      try {
-        Files.size(path);
-        // we shouldn't make it to this line. Not using thrown.expect because
-        // I still want to run a few lines after the exception.
-        assertThat(false).isTrue();
-      } catch (NoSuchFileException nsf) {
-        // that's what we wanted, we're good.
-      }
+      // channel now closed, the file should be there and with the new contents.
+      assertThat(Files.exists(path)).isTrue();
+      assertThat(Files.size(path)).isGreaterThan(100L);
     }
-    // channel now closed, the file should be there and with the new contents.
-    assertThat(Files.exists(path)).isTrue();
-    assertThat(Files.size(path)).isGreaterThan(100L);
   }
 
   @Test
   public void testWrite_trailingSlash() throws IOException {
-    thrown.expect(CloudStoragePseudoDirectoryException.class);
-    Files.write(Paths.get(URI.create("gs://greenbean/adipose/")), FILE_CONTENTS, UTF_8);
+    try (FileSystem fs = helper.forBucket("greenbean")) {
+      thrown.expect(CloudStoragePseudoDirectoryException.class);
+      Files.write(fs.getPath("/adipose/"), FILE_CONTENTS, UTF_8);
+    }
   }
 
   @Test
   public void testExists() throws IOException {
-    assertThat(Files.exists(Paths.get(URI.create("gs://military/fashion")))).isFalse();
-    Files.write(Paths.get(URI.create("gs://military/fashion")), "(✿◕ ‿◕ )ノ".getBytes(UTF_8));
-    assertThat(Files.exists(Paths.get(URI.create("gs://military/fashion")))).isTrue();
+    try (FileSystem fs = helper.forBucket("military")) {
+      assertThat(Files.exists(fs.getPath("/fashion"))).isFalse();
+      Files.write(fs.getPath("/fashion"), "(✿◕ ‿◕ )ノ".getBytes(UTF_8));
+      assertThat(Files.exists(fs.getPath("/fashion"))).isTrue();
+    }
   }
 
   @Test
-  public void testExists_trailingSlash() {
-    assertThat(Files.exists(Paths.get(URI.create("gs://military/fashion/")))).isTrue();
-    assertThat(Files.exists(Paths.get(URI.create("gs://military/fashion/.")))).isTrue();
-    assertThat(Files.exists(Paths.get(URI.create("gs://military/fashion/..")))).isTrue();
+  public void testExists_trailingSlash() throws IOException {
+    try (FileSystem fs = helper.forBucket("military")) {
+      assertThat(Files.exists(fs.getPath("/fashion/"))).isTrue();
+      assertThat(Files.exists(fs.getPath("/fashion/."))).isTrue();
+      assertThat(Files.exists(fs.getPath("/fashion/.."))).isTrue();
+    }
   }
 
   @Test
-  public void testExists_trailingSlash_disablePseudoDirectories() {
-    try (CloudStorageFileSystem fs = forBucket("military", usePseudoDirectories(false))) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method names should not contain underscores](https://www.codacy.com/app/mziccard/gcloud-java/file/2333823942/issues/source?bid=3123107&fileBranchId=3190440#l451)